pywhy_graphs/algorithms/cg.py

```diff
@@ -0,0 +1,86 @@
+import copy
+from collections import OrderedDict, deque
+
+from pywhy_graphs import CG
+
+__all__ = ["is_valid_cg"]
+
+
+def is_valid_cg(graph: CG):
+    """
+    Checks if a supplied chain graph is valid.
+
+    This implements the original definition of a (Lauritzen Wermuth Frydenberg) chain graph as
+    presented in [1]_.
+
+    Define a cycle as a series of nodes X_1 -o X_2 ... X_n -o X_1 where the edges may be directed or
+    undirected. Note that directed edges in a cycle must all be aligned in the same direction. A
+    chain graph may only contain cycles consisting of only undirected edges. Equivalently, a chain
+    graph does not contain any cycles with one or more directed edges.
+
+    Parameters
+    __________
+    graph : CG
+        The graph.
+
+    Returns
+    _______
+    is_valid : bool
+        Whether supplied `graph` is a valid chain graph.
+
+    References
+    ----------
+    .. [1] Frydenberg, Morten. “The Chain Graph Markov Property.” Scandinavian Journal of
+    Statistics, vol. 17, no. 4, 1990, pp. 333–53. JSTOR, http://www.jstor.org/stable/4616181.
+    Accessed 15 Apr. 2023.
+
+
+    """
+
+    # Check if directed edges are acyclic
+    undirected_edge_name = graph.undirected_edge_name
+    directed_edge_name = graph.directed_edge_name
+    all_nodes = graph.nodes()
+    G_undirected = graph.get_graphs(edge_type=undirected_edge_name)
+    G_directed = graph.get_graphs(edge_type=directed_edge_name)
+
+    # Search over all nodes.
+    for v in all_nodes:
+        queue = deque([])
```

Comment:
Add a type annotation to `queue` to fix the mypy complaint

Reply:
Can do - also I have the pre-commit hooks PR which is open for review which will help me catch this :)))